sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -938,6 +939,14 @@ case class MapObjects private(
         executeFuncOnCollection(input).foreach(builder += _)
         mutable.ArraySeq.make(builder.result())
       }
+    case Some(cls) if classOf[immutable.ArraySeq[_]].isAssignableFrom(cls) =>
```

Comment:
not related to this PR, but this `MapObjects` expression gets more and more complicated. Can we rewrite it with `RuntimeReplacable` using `StaticInvoke/Invoke` so that we can implement it with pure scala?

Reply:
Okay, let me give it a try.